pkg/controllers/dynakube/injection/conditions.go

```diff
@@ -32,3 +46,80 @@ func setCodeModulesInjectionCreatedCondition(conditions *[]metav1.Condition) {
 	}
 	_ = meta.SetStatusCondition(conditions, condition)
 }
+
+func setNamespacesMonitoredSelectorCondition(conditions *[]metav1.Condition, selectorType string, configured bool, names []string) {
+	var condType string
+
+	switch selectorType {
+	case "OneAgent":
+		condType = oneAgentNamespacesMonitoredConditionType
+	case "MetadataEnrichment":
+		condType = metadataEnrichmentNamespacesMonitoredConditionType
+	}
+
+	cond := metav1.Condition{Type: condType}
+
+	switch {
+	case !configured:
+		_ = meta.RemoveStatusCondition(conditions, condType)
+
+		return
+	case len(names) == 0:
+		cond.Status = metav1.ConditionFalse
+		cond.Reason = noMatchesReason
+		cond.Message = "0 namespaces match"
+	default:
+		cond.Status = metav1.ConditionTrue
+		cond.Reason = matchesFoundReason
+		msg := formatMatchMessage(names, maxNamesInMsg)
+		cond.Message = msg
+	}
+
+	cond.LastTransitionTime = metav1.Now()
+	_ = meta.SetStatusCondition(conditions, cond)
+}
+
+func updateCollectedNamespacesMonitoredCondition(conditions *[]metav1.Condition) {
+	oa := meta.FindStatusCondition(*conditions, oneAgentNamespacesMonitoredConditionType)
+	me := meta.FindStatusCondition(*conditions, metadataEnrichmentNamespacesMonitoredConditionType)
+
+	collected := metav1.Condition{
+		Type:               namespacesMonitoredConditionType,
+		LastTransitionTime: metav1.Now(),
+	}
+
+	if oa != nil && oa.Status == metav1.ConditionTrue || me != nil && me.Status == metav1.ConditionTrue {
+		collected.Status = metav1.ConditionTrue
+		collected.Reason = matchesFoundReason
+		collected.Message = "At least one selector has matching namespaces"
+		_ = meta.SetStatusCondition(conditions, collected)
+
+		return
+	}
+
+	if oa != nil && oa.Reason == selectorNotConfiguredReason && me != nil && me.Reason == selectorNotConfiguredReason {
+		collected.Status = metav1.ConditionFalse
+		collected.Reason = selectorNotConfiguredReason
+		collected.Message = "No selectors configured"
+		_ = meta.SetStatusCondition(conditions, collected)
+
+		return
+	}
```

Comment:
This being a `false` status gives me the impression that "nothing matches" the selector, but the opposite is true.

I would be for to just not have this case, just have "we have matches, look at them" and "we dont have matches, something maybe off".

It would simplify the logic a bunch.